### ISSUE TITLE
add checks for early references to global names

### ISSUE THREAD

**Issue by nicks** [google/closure-compiler#583]
this class of error seems to hit me a bunch:

``` js
// ==ClosureCompiler==
// @compilation_level ADVANCED_OPTIMIZATIONS
// @output_file_name default.js
// ==/ClosureCompiler==

var x = {};
alert(x.y);
x.y = 3;
```

closure-compiler does not warn about this today, but I think it should. I was thinking about either moving the`VariableReferenceCheck` after `CollapseProperties`, or doing something clever in `CheckGlobalNames`.

any thoughts or advice on this?
